coffee/lib/abstract-chosen.coffee

```diff
@@ -178,13 +179,29 @@ class AbstractChosen
         option.search_text = if option.group then option.label else option.html
 
         unless option.group and not @group_search
-          option.search_match = this.search_string_match(option.search_text, regex)
+          option.search_match = this.search_string_match(this.normalize_search_text(option.search_text), regex)
           results += 1 if option.search_match and not option.group
 
           if option.search_match
             if searchText.length
-              startpos = option.search_text.search zregex
-              text = option.search_text.substr(0, startpos + searchText.length) + '</em>' + option.search_text.substr(startpos + searchText.length)
+              startpos = this.normalize_search_text(option.search_text).search zregex
+              poslength = this.normalize_search_text(searchText).length
+
+              # If any normalization changed the search_text, perform a better
+              # logic to highlight results
+              if this.normalize_search_text(option.search_text) != option.search_text
+                for i in [0..option.search_text.length - 1] by 1
+                  if new RegExp(this.normalize_search_text(searchText), 'i').test(this.normalize_search_text(option.search_text.substr(i))) is false
```

Comment:
btw, you should cache ``this.normalize_search_text(searchText)`` (and its escaping, and potentially even the regex itself) in a local variable instead of normalizing again for each iteration. This will be faster